src/core/server/http/integration_tests/lifecycle.test.ts

```diff
@@ -222,6 +223,39 @@ describe('OnPreAuth', () => {
 
     await supertest(innerServer.listener).get('/').expect(200, { customField: 'undefined' });
   });
+
+  it('has no access to request body', async () => {
+    const { registerOnPreAuth, server: innerServer, createRouter } = await server.setup(setupDeps);
+    const router = createRouter('/');
+    let requestBody = null;
+    registerOnPreAuth((req, res, t) => {
+      requestBody = req.body;
+      return t.next();
+    });
```

Comment:
optional:

```ts
expect.assertions(1);

registerOnPreAuth((req, res, t) => {
      expect(req.body).toStrictEqual({});
      return t.next();
});
```

Reply:
that will return `500 Internal error` from the server without any meaningful error message